### PR TITLE
Fix edge case when casting gluon Block before export

### PR DESCRIPTION
**Pull request by leezu**
Fixes https://github.com/apache/incubator-mxnet/issues/18843

### CODE REVIEW

File: python/mxnet/gluon/parameter.py [hunk 1]
```diff
@@ -662,6 +662,7 @@ def cast(self, dtype):
             The new data type.
         """
         self._dtype = dtype
+        self._var = None  # Clear Symbol Variable as it caches the dtype
```
Comment (szha) [inline python/mxnet/gluon/parameter.py:665]: how is this related to the cached graph and cached op fields?

Comment (leezu) [reply python/mxnet/gluon/parameter.py:665]: The _var is used as input

**Comment by leezu** [comment]
@mxnet-bot run ci [centos-gpu, unix-gpu]

**Comment by szha** [comment]
cc @sandeep-krishnamurthy in case team needs to backport this fix.
